math/fastor/include/algebra/math/impl/fastor_matrix.hpp

```diff
@@ -0,0 +1,123 @@
+/** Algebra plugins, part of the ACTS project
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "algebra/math/common.hpp"
+#include "algebra/qualifiers.hpp"
+
+// Fastor include(s).
+#include <Fastor/Fastor.h>
+
+// System include(s).
+#include <cstddef>  // for the std::size_t type
+
+namespace algebra::fastor::math {
+
+/// "Matrix actor", assuming an Fastor matrix
+template <typename scalar_t>
+struct actor {
+
+  /// Size type
+  using size_ty = std::size_t;
+
+  /// Scalar type
+  using scalar_type = scalar_t;
+
+  /// 2D matrix type
+  template <int ROWS, int COLS>
+  using matrix_type = algebra::fastor::matrix_type<scalar_t, ROWS, COLS>;
+
+  /// Array type
+  template <size_ty N>
+  using array_type = storage_type<scalar_type, N>;
+
+  /// 3-element "vector" type
+  using vector3 = array_type<3>;
+
+  /// Operator getting a reference to one element of a non-const matrix
+  template <int ROWS, int COLS>
+  ALGEBRA_HOST_DEVICE inline scalar_t &element(matrix_type<ROWS, COLS> &m,
+                                               int row, int col) const {
+    return m(row, col);
+  }
+
+  /// Operator getting one value of a const matrix
+  template <int ROWS, int COLS>
+  ALGEBRA_HOST_DEVICE inline scalar_t element(const matrix_type<ROWS, COLS> &m,
+                                              int row, int col) const {
+    return m(row, col);
+  }
+
+  /// Operator getting a block of a const matrix
+  template <size_ty ROWS, size_ty COLS, class input_matrix_type>
+  ALGEBRA_HOST_DEVICE matrix_type<ROWS, COLS> block(const input_matrix_type &m,
+                                                    size_ty row, size_ty col) {
+    // In Fastor::fseq, the last element is not included.
+    return m(Fastor::fseq<row, row + ROWS>(), Fastor::fseq<col, col + COLS>());
+  }
+
+  /// Operator setting a block
+  template <size_ty ROWS, size_ty COLS, class input_matrix_type>
+  ALGEBRA_HOST_DEVICE void set_block(input_matrix_type &m,
+                                     const matrix_type<ROWS, COLS> &b,
+                                     size_ty row, size_ty col) {
+    m(Fastor::fseq<row, row + ROWS>(), Fastor::fseq<col, col + COLS>()) = b;
+  }
+
+  // Create zero matrix
+  template <int ROWS, int COLS>
```

Comment:
I think these should be ```size_ty``` (not sure why it is called that), could you check please?

Reply:
Good catch. I changed it to `size_ty`.